### PR TITLE
Private links counter in the header

### LINKED ISSUE #247: Statistics : number private links, database and cache size
Discussed in #106:

  * Add counters for number of public and private links to the tools page.
  * Add counters for datastore and cache directories to the tools page.

Maybe highlight cache size if it's anormaly big.

### PR DESCRIPTION
**Pull request by ArthurHoaro**
Also moved PageBuilder to its own file to clean up `index.php`.

Fixes #247

### CODE REVIEW

File: tpl/page.header.html [hunk 1]
```diff
@@ -2,7 +2,8 @@
 <div id="logo" title="Share your links !" onclick="document.location='?';"></div>
 
 <div id="linkcount" class="nomobile">
-    {if="!empty($linkcount)"}{$linkcount} links{/if}
+    {if="!empty($linkcount)"}{$linkcount} links{/if}<br>
+    {if="!empty($privateLinkcount)"}{$privateLinkcount} privates{/if}
```
Comment (virtualtam) [inline tpl/page.header.html]: s/privates/private links/

this is a subtle one for non-English speakers, as "privates" is a way of designating one's genitalia - http://www.freethesaurus.com/privates

File: application/PageBuilder.php [hunk 2]
```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * Class PageBuilderThis class is in charge of building the final page.
```
Comment (virtualtam) [inline application/PageBuilder.php]: extra `Class PageBuilder`

File: application/PageBuilder.php [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+     * usage : $PAGE->render404('The link was deleted')
+     *
+     * @param string $message A messate to display what is not found
+     */
+    public function render404($message='The page you are trying to reach does not exist or has been deleted.') {
+        header($_SERVER['SERVER_PROTOCOL'] . ' 404 Not Found');
+        $this->tpl->assign('error_message', $message);
+        $this->renderPage('404');
+    }
+}
```
Comment (virtualtam) [inline application/PageBuilder.php]: gitattributes don't seem that efficient at checking carriage returns :(

File: application/PageBuilder.php [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+ * p = new PageBuilder();
+ * p.assign('myfield','myvalue');
+ * p.renderPage('mytemplate');
+ */
+class PageBuilder
+{
+    /**
+     * @var RainTPL RainTPL instance.
+     */
+    private $tpl; //
```
Comment (virtualtam) [inline application/PageBuilder.php]: empty comment

File: tpl/page.header.html (hunk 1 above)
Comment (ArthurHoaro) [inline tpl/page.header.html]: Isn't it exactly why one would save links in private mode? :)

**Pull request merged**